File: Data Structures & Algorithms/concatenation-of-array/submission-6.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        n = len(nums)
        newArr = [0] * (n * 2)

        for i in range(n):
            newArr[i] = newArr[i + n] = nums[i]

        return newArr

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const numsSet = new Set(nums)
        let longest = 0

        for (const num of nums) {
            if (numsSet.has(num - 1)) {
                continue
            }
            
            let curr = num
            let length = 1
            while (numsSet.has(curr + 1)) {
                length ++
                curr ++
            }

            longest = Math.max(longest, length)
        }

        return longest
    }
}
